assets/js/admin.js

```diff
@@ -0,0 +1,73 @@
+/* global ajaxurl, slaAdmin */
+(function ($) {
+    $(function () {
+        function slaAdminAction() {
+            const $sla = {
+                clearCache: $('#clear_cache_btn'),
+            };
+
+            // Spinner button.
+            const spinnerButton = '<span class="spinner is-active"></span>';
+
+            // Bind events.
+            const bindEvents = function () {
+                $sla.clearCache.on('click', function (e) {
+                    e.preventDefault();
+                    $sla.clearCache.addClass( 'disabled' );
+                    var data = {
+                        action: 'sla_clear_user_cache',
+                        nonce: slaAdmin.nonce,
+                    }
+                    $sla.clearCache.append(spinnerButton);
+                    processStep(1, data, $sla.clearCache);
+                });
+            };
+
+            /**
+             * Process steps.
+             *
+             * @param {Number} step For batch wise updates.
+             * @param {Object} data Data to pass to ajax request.
+             * @param {HTMLElement} el Spinner button.
+             */
+            function processStep(step, data, el) {
+                data.step = step;
+                $.ajax({
+                    url    : ajaxurl,
+                    dataType: 'json',
+                    data   : data,
+                    method : 'POST',
+                    success: function (response) {
+                        if (response.success) {
+                            if (response.data.step === 'done') {
+                                el.find('span').remove();
+                                $sla.clearCache.removeClass( 'disabled' );
+                            } else {
+                                processStep(parseInt(response.data.step, 10), data, el);
+                            }
+                            return false;
+                        } else {
+                            alert(slaAdmin.error);
+                            el.find('span').remove();
+                            $sla.clearCache.removeClass( 'disabled' );
+                        }
+                    },
+                    error  : function () {
+                        alert(slaAdmin.error);
```

Comment:
Instead of `alert`, can we print the error message below the clear cache button? By doing so we can display more detailed errors. We can even add the status there too, for example:

![image](https://user-images.githubusercontent.com/5423135/145017280-dc2bf074-12ed-4c95-97b1-7549b0311e9a.png)

Reply:
Added the message span below the button, demo https://cln.sh/u5NhwT